### PR TITLE
Refactor ai legs TM-3248

### PR DESCRIPTION
**Pull request by elizabeth-jimenez**
Current Status: Ready for Review

Wow, what a beautiful pic 💟 
image

AgendaItemLegs has been simplified since it now is only handling displaying of legs.

### CODE REVIEW

**Comment by elizabeth-jimenez** [comment]
Before:
image

After:
image

**Comment by elizabeth-jimenez** [comment]
Current known issues to be addressed in other tickets:
- Check v2/positions to return language requirements for position and add to api to print
- When clicking outside of TED, close calendar
- Have TED open outside of the leg col
- Fix gaps that occassionally appear on Leg borders
- On hover highlight the row

File: src/Components/Agenda/AgendaItemLegsForm/AgendaItemLegsForm.jsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+  useEffect(() => {
+    setNumLegs(legs.length);
+  }, [legs]);
+
+  return (
+    <>
+      {
+        legsLoading &&
+          <Spinner type="legs" size="small" />
```
Comment (scott062) [inline src/Components/Agenda/AgendaItemLegsForm/AgendaItemLegsForm.jsx:57]: indent?

File: src/Components/Agenda/AgendaLeg/AgendaLeg.jsx [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+  const formatDate = (d) => d && isDate(new Date(d)) ? format(new Date(d), 'MM/dd/yy') : '';
+
+  const getCalendar = () => (
+      <>
+        {formatDate(TED)}
+        <FA name="calendar" style={{ color: `${calendarHidden ? 'black' : 'red'}` }} onClick={() => setCalendarHidden(!calendarHidden)} />
+        {
+          !calendarHidden &&
+            <div className="ted-calendar-container" id={`cal-${legNum}`}>
```
Comment (scott062) [inline src/Components/Agenda/AgendaLeg/AgendaLeg.jsx:57]: && 
next line indent or not - up to you but just want to be consistent 

To-Do: Followup PR

File: src/Components/Agenda/AgendaItemLegsForm/AgendaItemLegsForm.jsx (hunk 1 above)
Comment (scott062) [reply src/Components/Agenda/AgendaItemLegsForm/AgendaItemLegsForm.jsx:57]: See comment below

**Comment by scott062** [comment]
To-Do: AIH card vs row (space on right side) 
Adjust margins and width of cards to match content borders more closely

**Comment by SidelineCory24** [comment]
nice update
image

**Review by scott062 — APPROVED** [review]
Will address to-dos and feedback in next PR forked off of this. Good to go for now. 👩‍🎤

**Pull request merged**